### ISSUE TITLE
[jiva] Mysql sts with openebs jiva struck into Init State as failed to mount volume

### ISSUE THREAD

**Issue by ismail0077** [openebs/openebs#2440]
<h2>Description</h2>
<p>Mysql sts with OpenEBS jiva struck into <b>inti state</b> as failed to mount the volumes. Volumes were in a healthy state checked exec into maya-api-server. Restarting Mysql got on schedule with the same node. </p>

<p>Worked around with cluster node with session logging out and loggin in <b>iscsiadm -m session -P 3</b> But volume mount failed to schedule on cluster node even after pod restart. 
 </p>

<p>dmesg of Cluster Node had this error from Initiator side <b>LUN larger than allowed by the host adapter</b> (complete logs are mentioned).</p>

<h2>Environment</h2>

**Cluster Details**

- Application running on 4 node cluster - 1.10.11-gke.1
- Openebs version 0.8.0

**Resource Limit of Mysql pod**

-    CPU: 500m
-    memory: 1Gi

**Storage Class**

- openebs-jiva-default
- Request Storage - 10Gi
- Replica Count - 3

<h2>Log Details</h2>

`kubectl get po -n dev`
```
NAME                                                             READY     STATUS             RESTARTS   AGE       IP
mysql-0                                                          0/2       Init:0/2           0          10s       <none>    
```
**Mysql pod Events**

`kubectl describe po -n dev mysql-0`
```
Events:
  Type     Reason                  Age   From                                                          Message
  ----     ------                  ----  ----                                                          -------
  Normal   Scheduled               2m    default-scheduler                                             Successfully assigned mysql-0 to gke-node1-cluster-default-pool-cc65d8e5-fcjw
  Normal   SuccessfulAttachVolume  2m    attachdetach-controller                                       AttachVolume.Attach succeeded for volume "pvc-456bfbb7-2b87-11e9-b132-42010a8000e3"
  Normal   SuccessfulMountVolume   2m    kubelet, gke-node1-cluster-default-pool-cc65d8e5-fcjw  MountVolume.SetUp succeeded for volume "conf"
  Normal   SuccessfulMountVolume   2m    kubelet, gke-node1-cluster-default-pool-cc65d8e5-fcjw  MountVolume.SetUp succeeded for volume "config-map"
  Normal   SuccessfulMountVolume   2m    kubelet, gke-node1-cluster-default-pool-cc65d8e5-fcjw  MountVolume.SetUp succeeded for volume "default-token-gntdt"
  Warning  FailedMount             12s   kubelet, gke-node1-cluster-default-pool-cc65d8e5-fcjw  Unable to mount volumes for pod "mysql-0_dev(80966e87-44c1-11e9-b132-42010a8000e3)": timeout expired waiting for volumes to attach/mount for pod "dev"/"mysql-0". list of unattached/unmounted volumes=[maya-database]
```
**Controller Pod Logs**

`kubectl logs -f pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-ctrl-78f44768bf-bkxlt -n dev -c pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-ctrl-con`
```
time="2019-03-12T11:11:08Z" level=warning msg=busy
time="2019-03-12T11:11:08Z" level=info msg="Create Replica for address tcp://10.8.6.16:9502"
time="2019-03-12T11:11:08Z" level=info msg="check if replica tcp://10.8.6.16:9502 is already added"
time="2019-03-12T11:11:08Z" level=info msg="check if any WO replica available"
time="2019-03-12T11:11:08Z" level=warning msg="can have only one WO replica at a time, found WO replica: %stcp://10.8.5.164:9502"
time="2019-03-12T11:11:08Z" level=error msg="Error in request: can only have one WO replica at a time, found WO Replica: tcp://10.8.5.164:9502"
10.8.6.16 - - [12/Mar/2019:11:11:08 +0000] "POST /v1/replicas HTTP/1.1" 500 225
time="2019-03-12T11:11:09Z" level=error msg="Mode: ReadOnly"
time="2019-03-12T11:11:09Z" level=error msg="Mode: ReadOnly"
time="2019-03-12T11:11:09Z" level=error msg="Mode: ReadOnly"
time="2019-03-12T11:11:09Z" level=warning msg=busy
time="2019-03-12T11:11:10Z" level=error msg="Mode: ReadOnly"
time="2019-03-12T11:11:10Z" level=error msg="Mode: ReadOnly"
time="2019-03-12T11:11:10Z" level=error msg="Mode: ReadOnly"
```
**Node-1 Logs**
`dmesg`
```
[  613.733026] sd 1:0:0:0: lun278722843558035 has a LUN larger than allowed by the host adapter
[  619.932423] EXT4-fs (sda1): re-mounted. Opts: (null)
[  620.008599] EXT4-fs (sda1): re-mounted. Opts: (null)
[  680.186474] scsi host1: iSCSI Initiator over TCP/IP
[  680.446890] scsi 1:0:0:0: Direct-Access     CLOUDBYT OPENEBS          0.2  PQ: 0 ANSI: 5
[  680.447924] sd 1:0:0:0: Attached scsi generic sg9 type 0
[  680.451594] sd 1:0:0:0: [sdj] 20971520 512-byte logical blocks: (10.7 GB/10.0 GiB)
[  680.451597] sd 1:0:0:0: [sdj] 4096-byte physical blocks
[  680.451786] sd 1:0:0:0: [sdj] Write Protect is off
[  680.451789] sd 1:0:0:0: [sdj] Mode Sense: 03 00 10 08
[  680.451976] sd 1:0:0:0: [sdj] No Caching mode page found
[  680.457555] sd 1:0:0:0: [sdj] Assuming drive cache: write through
[  680.550744] sd 1:0:0:0: [sdj] Attached SCSI disk
[  736.225485] sd 1:0:0:0: lun280922252615716 has a LUN larger than allowed by the host adapter
[  858.626139] sd 1:0:0:0: lun277624971587619 has a LUN larger than allowed by the host adapter
[  981.077580] sd 1:0:0:0: lun277626444157020 has a LUN larger than allowed by the host adapter
[ 1103.634882] sd 1:0:0:0: lun278723981447335 has a LUN larger than allowed by the host adapter
[ 1139.514279] JBD2: Clearing recovery information on journal
[ 1157.992780] EXT4-fs (sdj): warning: mounting fs with errors, running e2fsck is recommended
[ 1158.029412] EXT4-fs (sdj): mounted filesystem with ordered data mode. Opts: (null)
[ 1226.145788] sd 1:0:0:0: lun278723634569344 has a LUN larger than allowed by the host adapter
[ 1228.772917] EXT4-fs (sdj): mounted filesystem with ordered data mode. Opts: (null)
```
`fdisk -l`
```
Disk /dev/sdj: 10 GiB, 10737418240 bytes, 20971520 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes
```
`journalctl -u kubelet | grep pvc-456bfbb7-2b87-11e9-b132-42010a8000e3`
```
Mar 12 12:41:27 gke-maya-staging-cluster-default-pool-cc65d8e5-fcjw kubelet[2631]: E0312 12:41:27.674565    2631 iscsi_util.go:338] iscsi: failed to mount iscsi volume /dev/disk/by-path/ip-10.11.253.88:3260-iscsi-iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-lun-0 [ext4] to /var/lib/kubelet/plugins/kubernetes.io/iscsi/iface-default/10.11.253.88:3260-iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-lun-0, error 'fsck' found errors on device /dev/disk/by-path/ip-10.11.253.88:3260-iscsi-iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-lun-0 but could not correct them: fsck from util-linux 2.27.1
Mar 12 12:41:27 gke-maya-staging-cluster-default-pool-cc65d8e5-fcjw kubelet[2631]: E0312 12:41:27.674772    2631 nestedpendingoperations.go:263] Operation for "\"kubernetes.io/iscsi/10.11.253.88:3260:iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3:0\"" failed. No retries permitted until 2019-03-12 12:41:43.674712248 +0000 UTC m=+7014865.960275925 (durationBeforeRetry 16s). Error: "MountVolume.WaitForAttach failed for volume \"pvc-456bfbb7-2b87-11e9-b132-42010a8000e3\" (UniqueName: \"kubernetes.io/iscsi/10.11.253.88:3260:iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3:0\") pod \"mysql-0\" (UID: \"e08cb475-44c3-11e9-b132-42010a8000e3\") : 'fsck' found errors on device /dev/disk/by-path/ip-10.11.253.88:3260-iscsi-iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-lun-0 but could not correct them: fsck from util-linux 2.27.1\n/dev/sdi: Superblock needs_recovery flag is clear, but journal has data.\n/dev/sdi: Run journal anyway\n\n/dev/sdi: UNEXPECTED INCONSISTENCY; RUN fsck MANUALLY.\n\t(i.e., without -a or -p options)\n."
Mar 12 12:41:43 gke-maya-staging-cluster-default-pool-cc65d8e5-fcjw kubelet[2631]: I0312 12:41:43.766352    2631 reconciler.go:262] operationExecutor.MountVolume started for volume "pvc-456bfbb7-2b87-11e9-b132-42010a8000e3" (UniqueName: "kubernetes.io/iscsi/10.11.253.88:3260:iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3:0") pod "mysql-0" (UID: "e08cb475-44c3-11e9-b132-42010a8000e3")
Mar 12 12:41:43 gke-maya-staging-cluster-default-pool-cc65d8e5-fcjw kubelet[2631]: I0312 12:41:43.766486    2631 operation_generator.go:447] MountVolume.WaitForAttach entering for volume "pvc-456bfbb7-2b87-11e9-b132-42010a8000e3" (UniqueName: "kubernetes.io/iscsi/10.11.253.88:3260:iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3:0") pod "mysql-0" (UID: "e08cb475-44c3-11e9-b132-42010a8000e3") DevicePath ""
Mar 12 12:41:44 gke-maya-staging-cluster-default-pool-cc65d8e5-fcjw kubelet[2631]: E0312 12:41:44.181706    2631 iscsi_util.go:338] iscsi: failed to mount iscsi volume /dev/disk/by-path/ip-10.11.253.88:3260-iscsi-iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-lun-0 [ext4] to /var/lib/kubelet/plugins/kubernetes.io/iscsi/iface-default/10.11.253.88:3260-iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-lun-0, error 'fsck' found errors on device /dev/disk/by-path/ip-10.11.253.88:3260-iscsi-iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-lun-0 but could not correct them: fsck from util-linux 2.27.1
Mar 12 12:41:44 gke-maya-staging-cluster-default-pool-cc65d8e5-fcjw kubelet[2631]: E0312 12:41:44.181850    2631 nestedpendingoperations.go:263] Operation for "\"kubernetes.io/iscsi/10.11.253.88:3260:iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3:0\"" failed. No retries permitted until 2019-03-12 12:42:16.18179006 +0000 UTC m=+7014898.467353731 (durationBeforeRetry 32s). Error: "MountVolume.WaitForAttach failed for volume \"pvc-456bfbb7-2b87-11e9-b132-42010a8000e3\" (UniqueName: \"kubernetes.io/iscsi/10.11.253.88:3260:iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3:0\") pod \"mysql-0\" (UID: \"e08cb475-44c3-11e9-b132-42010a8000e3\") : 'fsck' found errors on device /dev/disk/by-path/ip-10.11.253.88:3260-iscsi-iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3-lun-0 but could not correct them: fsck from util-linux 2.27.1\n/dev/sdi: Superblock needs_recovery flag is clear, but journal has data.\n/dev/sdi: Run journal anyway\n\n/dev/sdi: UNEXPECTED INCONSISTENCY; RUN fsck MANUALLY.\n\t(i.e., without -a or -p options)\n."
Mar 12 12:44:36 gke-maya-staging-cluster-default-pool-cc65d8e5-fcjw kubelet[2806]: I0312 12:44:36.892034    2806 reconciler.go:262] operationExecutor.MountVolume started for volume "pvc-456bfbb7-2b87-11e9-b132-42010a8000e3" (UniqueName: "kubernetes.io/iscsi/10.11.253.88:3260:iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3:0") pod "mysql-0" (UID: "8b2758bd-44c4-11e9-b132-42010a8000e3")
Mar 12 12:44:36 gke-maya-staging-cluster-default-pool-cc65d8e5-fcjw kubelet[2806]: I0312 12:44:36.892136    2806 operation_generator.go:447] MountVolume.WaitForAttach entering for volume "pvc-456bfbb7-2b87-11e9-b132-42010a8000e3" (UniqueName: "kubernetes.io/iscsi/10.11.253.88:3260:iqn.2016-09.com.openebs.jiva:pvc-456bfbb7-2b87-11e9-b132-42010a8000e3:0") pod "mysql-0" (UID: "8b2758bd-44c4-11e9-b132-42010a8000e3") DevicePath ""
```
<h2>Possible Solution</h2>

<p>Manually mounted the disk to new folder and unmounted it back, where pod got back to running state from Init state.</p>

- `fdisk -l`
- `mkdir folder`
- `mount /dev/<disk-name> folder`
- `umount folder`

<h2>Your Environment</h2>

- `kubectl get nodes`:
- `kubectl get pods --all-namespaces`:
- `kubectl get services`:
- `kubectl get sc`:
- `kubectl get pv`:
- `kubectl get pvc`:
- OS (from `/etc/os-release`):
- Kernel (from `uname -a`):
- Install tools:
- Others:
OpenEBS version : 0.8.0

<h3>Controller and Replica Logs Attachments:</h3>

[log1.txt](https://github.com/openebs/openebs/files/3003767/log1.txt)
[log2.txt](https://github.com/openebs/openebs/files/3003772/log2.txt)
[log3.txt](https://github.com/openebs/openebs/files/3003774/log3.txt)
[log4.txt](https://github.com/openebs/openebs/files/3003776/log4.txt)

**Comment by utkarshmani1997** [comment]
@ismail0077 this issue was observed on 0.8.0 where replica got stuck after read/write timeout, due to which replica never get reconnected with controller so in this case volume went into RO since quorum is not met and the other replicas never connected back. This issue has been fixed in 0.8.1.
Ref: https://github.com/openebs/jiva/pull/168

**Comment by utkarshmani1997** [comment]
@ismail0077 this issue was observed on 0.8.0 where replica got stuck after read/write timeout, due to which replica never get reconnected with controller so in this case volume went into RO since quorum is not met and the other replicas never connected back. This issue has been fixed in 0.8.1.
Ref: https://github.com/openebs/jiva/pull/168
